### PR TITLE
extract runtime cache

### PR DESCRIPTION
**Pull request by bestform**
| Q             | A
| ------------- | ---
| Branch        | 7.1.x
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no 
| Deprecations? | no
| Tests pass?   | yes
| Fixed issues  | chameleon-system/chameleon-system#639
| License       | MIT

This is the prerequisite for extensions to be able to influence the way the snippet directory service caches its results.

### CODE REVIEW

File: src/ViewRendererBundle/objects/helper/TPkgViewRendererSnippetDirectoryRuntimeCache.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
+<?php
+
```
Comment (UlrichKu) [inline src/ViewRendererBundle/objects/helper/TPkgViewRendererSnippetDirectoryRuntimeCache.php:2]: It should have the license header.

File: src/ViewRendererBundle/objects/helper/TPkgViewRendererSnippetDirectoryRuntimeCache.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        static $cachedDir = null;
+        if (null !== $cachedDir) {
+            return $cachedDir;
+        }
+
+        $cachedDir = $this->subject->getSnippetBaseDirectory();
+
+        return $cachedDir;
+    }
+}
```
Comment (UlrichKu) [inline src/ViewRendererBundle/objects/helper/TPkgViewRendererSnippetDirectoryRuntimeCache.php:86]: Trailing empty line is missing (code formatter)?

File: src/ViewRendererBundle/objects/helper/TPkgViewRendererSnippetDirectoryRuntimeCache.php [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        if (isset($basePaths[$portalCacheId])) {
+            return $basePaths[$portalCacheId];
+        }
+
+        $basePaths[$portalCacheId] = $this->subject->getBasePaths($oPortal, $sBaseDirectory);
+
+        return $basePaths[$portalCacheId];
+    }
+
+    public function getSnippetBaseDirectory()
```
Comment (UlrichKu) [inline src/ViewRendererBundle/objects/helper/TPkgViewRendererSnippetDirectoryRuntimeCache.php:75]: it's not really necessary to cache this (only one if in the base implementation)

File: src/ViewRendererBundle/objects/helper/TPkgViewRendererSnippetDirectoryRuntimeCache.php [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    public function getBasePathsFromInstance($oPortal = null, $sBaseDirectory = null)
+    {
+        return $this->subject->getBasePathsFromInstance($oPortal, $sBaseDirectory);
+    }
+
+    public function getBasePaths($oPortal = null, $sBaseDirectory = null)
+    {
+        static $basePaths = array();
```
Comment (UlrichKu) [inline src/ViewRendererBundle/objects/helper/TPkgViewRendererSnippetDirectoryRuntimeCache.php:60]: -> []

File: src/ViewRendererBundle/objects/helper/TPkgViewRendererSnippetDirectoryRuntimeCache.php (hunk 1 above)
Comment (bestform) [reply src/ViewRendererBundle/objects/helper/TPkgViewRendererSnippetDirectoryRuntimeCache.php:2]: done

File: src/ViewRendererBundle/objects/helper/TPkgViewRendererSnippetDirectoryRuntimeCache.php (hunk 2 above)
Comment (bestform) [reply src/ViewRendererBundle/objects/helper/TPkgViewRendererSnippetDirectoryRuntimeCache.php:86]: done

File: src/ViewRendererBundle/objects/helper/TPkgViewRendererSnippetDirectoryRuntimeCache.php (hunk 3 above)
Comment (bestform) [reply src/ViewRendererBundle/objects/helper/TPkgViewRendererSnippetDirectoryRuntimeCache.php:75]: it is cached in the original implementation so I just extracted it as well to be consistent.

File: src/ViewRendererBundle/objects/helper/TPkgViewRendererSnippetDirectoryRuntimeCache.php (hunk 4 above)
Comment (bestform) [reply src/ViewRendererBundle/objects/helper/TPkgViewRendererSnippetDirectoryRuntimeCache.php:60]: done

**Pull request merged**